### DISCUSSION TITLE
can I go from mesh back to shape?

### DISCUSSION THREAD

**Discussion by tsdexter** [sgenoud/replicad#153] [General]
Is there a way to go from a mesh like `{triangles: [...], vertices: [...], normals: [...]}` back to a shape I can then perform more operations on such as `fillet`ing?

**Comment by sgenoud** [comment]
Unfortunately no - this is a feature that is far from trivial - you need to guess the topology of a part from its mesh. If someone is interested in working in this, [this post and its references](https://c3dlabs.com/blog/products/c3d-b-shaper-sdk-s-polygonal-mesh-to-boundary-representation-solid-conversion-algorithm-details-and--16/) might be a good starting point

**Reply by tsdexter** [reply]
Thanks, that's what I thought - just wanted to make sure I wasn't missing anything.
